gno.land/pkg/gnoland/app.go

```diff
@@ -89,11 +89,15 @@ func NewAppWithOptions(cfg *AppOptions) (abci.Application, error) {
 	baseApp.MountStoreWithDB(baseKey, dbadapter.StoreConstructor, cfg.DB)
 
 	// Construct keepers.
-	paramsKpr := params.NewParamsKeeper(mainKey, "vm")
+
+	km := params.NewPrefixKeyMapper()
+	km.RegisterPrefix(auth.ParamsPrefixKey)
+	km.RegisterPrefix(bank.ParamsPrefixKey)
+	km.RegisterPrefix(vm.ParamsPrefixKey)
```

Comment:
How about instead of PrefixKeyMapper, we call it KeeperMapper, and we register not just the prefix key, but we register the keeper itself?

e.g. auth, bank, vm could implement the following interface 
```go
type ParamfulKeeper interface {
    GetParamsKey() string
}
```

and km.RegisterKeepers(auth, bank, vm). 

This way, we can actually do a lookup in Params.SetBool (for example) to look up the keeper first by prefix.

See specific suggestion in https://github.com/gnolang/gno/pull/3176/files#r1908697754 which actually suggests removing PrefixKeyMapper/KeeperMapper entirely and just baking it into Params directly.

Reply:
done
 https://github.com/gnolang/gno/pull/3176#discussion_r1926137202